src/main/java/org/zanata/mt/dao/AbstractDAO.java

```diff
@@ -0,0 +1,22 @@
+package org.zanata.mt.dao;
+
+import java.io.Serializable;
+import javax.ejb.TransactionAttribute;
+import javax.persistence.EntityManager;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+public abstract class AbstractDAO<T> implements Serializable {
```

Comment:
If you decide to introduce DeltaSpike, you can use its facilities to build these objects.